src/api/app/helpers/webui/staging/workflow_helper.rb

```diff
@@ -0,0 +1,83 @@
+module Webui::Staging::WorkflowHelper
+  def build_progress(staging_project)
+    total = 0
+    final = 0
+
+    staging_project.building_repositories.each do |r|
+      total += r[:tobuild] + r[:final]
+      final += r[:final]
+    end
+
+    return 100 if total == 0
+    # if we have building repositories, make sure we don't exceed 99
+    [final * 100 / total, 99].min
+  end
+
+  def review_progress(staging_project)
+    staged_requests_numbers = staging_project.staged_requests.pluck(:number)
+    total = Review.where(bs_request: staging_project.staged_requests).count
```

Comment:
Use `length` instead of `count`!!! :see_no_evil: This is causing an extra unneeded query to the database

Reply:
In this specific case, it will always generate a query with `SELECT COUNT(*) ...`. So it's not an issue to use `.count`. However, I think we should always use `.size`. @bgeuken found this article which summarizes this "debate" on `.length`, `.count` and `.size`: https://medium.com/@craigsheen/count-vs-length-in-rails-4308e83f6292.

So now with this in mind, whatever `.size` uses under the hood will be the right one. In this case, it's `count` as `.size` generates a `SELECT COUNT(*) ...`.

We will use `.size`.